mkenv.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+import importlib
+import os
+import subprocess
+import sys
+
+# Requires at least Python 3.7. If placing this into a project that
+# requires a newer Python, you may elect to increase the version
+# requirement below.
+if sys.version_info[:2] < (3, 7):
+    raise SystemError('Only Python 3.7+ is allowed')
+
+MKENV_IMPL = 'mkenv_impl'
+HERE = os.path.dirname(os.path.abspath(__file__))
+MKENV_SUBDIR = '.mkenv'
+MKENV_REPO = os.environ.get('DL_MKENV_REPO', 'git@octocat.dlogics.com:datalogics/mkenv.git')
+MKENV_BRANCH = os.environ.get('DL_MKENV_BRANCH', 'main')
+DL_MKENV_ENVIRONMENT_OVERRIDE = 'DL_MKENV_REPO' in os.environ or 'DL_MKENV_BRANCH' in os.environ
+# The oldest Git v2 we have on our machines is 2.3, and it seems to work for mkenv.
+GIT_REQUIRED = (2, 3)
+
+
+def run(command_args, verbose=False, check=True, capture_output=False, *args, **kwargs):
+    if verbose:
+        print(' '.join(command_args), file=sys.stderr)
+    redirect_stderr = subprocess.PIPE if capture_output else None
+    redirect_stdout = subprocess.PIPE if capture_output else sys.stderr
+    return subprocess.run(command_args, check=check, stdout=redirect_stdout, stderr=redirect_stderr,
+                          *args, **kwargs)
+
+
+def get_mkenv_impl_from_git():
+    old_sys_path = sys.path.copy()
+    try:
+        mkenv_dir = os.path.join(HERE, MKENV_SUBDIR)
+        # snipe the verbose flag from the argv to decide how chatty to be about the
+        # git commands
+        verbose = '-v' in sys.argv or '--verbose' in sys.argv
+
+        # Check git version
+        completed = run(['git', 'version'], capture_output=True, check=True)
+        git_version = completed.stdout.decode().strip().split()[2]
+        # Windows has string stuff after the first three numbers, hence the [:3]
+        git_version_split = [int(x) for x in git_version.split('.')[:3]]
+        if tuple(git_version_split[:len(GIT_REQUIRED)]) < GIT_REQUIRED:
+            ver_string = '.'.join(str(x) for x in GIT_REQUIRED)
+            sys.exit(f'*** Git version {ver_string} or newer required, found {git_version}; '
```

Comment:
These lines also changed in the move to GitHub cloud, see this PR: https://octocat.dlogics.com/datalogics/apdfl-cpp-utils/pull/84/files

I'd recommend just copying the `mkenv.py` from that project and using it here

Reply:
Added